### ISSUE TITLE
Better "visit" tracking

### ISSUE THREAD

**Issue by smudge** [smudge/sojourn#3]
There are two ways to track visits:

1. Assign a visit ID to each visit, and tag all requests/events with that visit id. This requires defining what a "unique visit" means for your product in advance, but allows your data to be more compact, since you can discard data about uninteresting requests if they are within the visit window.
2. Don't assign a visit ID in advance, but track interesting things as they happen and then group them into visits after the fact. This allows you to choose a different visit window and rerun your analysis, but requires that you store more data up front in case you decide to change your visit window later.

Sojourn takes the 2nd approach. Within that approach, there are two ways of collecting data:

1. Store data about every web request and/or page view. Lots of data, but lots of flexibility.
2. Only track requests if they meet a set of conditions that make them candidates for the start of a new visit.

Sojourn takes the 2nd approach here as well, because the first was too much data for the early implementation to handle. Right now this means that requests are marked as potential new visits when A) the visitor is totally new to the site, B) the visitor's referer is that of a different site, or C) the visitor's request has UTM parameters.

But what this doesn't capture is what happens when a visitor comes to the site, leaves for a long time, and comes back again.

Ideally we'd just start storing data about every web request, but if we can't do that, there should be a timeout between requests where any new requests should be marked as a potential new visit. This could be configurable but should default to something like a day or a week. The downside is that there is still potential to lose valuable data if you decide to change your visit window, but at least there is some flexibility beyond just having to choose one visit window and stick with it.

So, TLDR; Add a time-out between requests which would cause a new `!sojourning` event to be fired, even if there is no UTM or referer data that would otherwise prompt a new `!sojourning`.
